AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -40,39 +41,116 @@ def __init__(self, sc_editor):
+
+        return selection
+
     def __validate_new_variable(self, new_variable_type: str) -> None:
         """
-        function for checking the provided variable type for validity
+        function for checking the provided variable type validity
 
         returns: None
         """
         assert type(new_variable_type) is str, "Wrong parameter type provided. new_variable_type was not str"
         assert new_variable_type in VARIABLE_TYPES_DICT, \
             "Wrong type provided. new_variable_type is not a valid type"
 
-    def create_new_variable(self, new_variable_type: str) -> None:
+    def __click_on_add_variable_button(self) -> None:
         """
-        Function for adding a new variable to the variable manager's list
-
-        param new_variable_type: the type of variable you want to create provided as a string. Is case-sensitive!
+        helper function for adding a new variable. Clicks on the 'Create Variable' button and waits for the UI to render
 
-        returns: None
+        returns None
         """
-        self.__validate_new_variable(new_variable_type)
-
         add_new_variable_button = self.variable_manager.findChild(QtWidgets.QPushButton, ADD_BUTTON_QT)
         add_new_variable_button.click()  # Click on Create Variable button
 
         helper.wait_for_condition((
             lambda: self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT) is not None),
             WAIT_TIME_SEC_3)
 
+    def __select_variable_type_from_create_menu(self, new_variable_type: str) -> None:
+        """
+        helper function for adding a new variable. selects the desired variable type from the menu that renders after
+        the create variable button is clicked. waits for the UI to respond after making a selection
+
+        params: The type of variable to create as a string. ie "Boolean"
+
+        returns Qwidget with the graph variables that have been created. Used for further helper methods
+        """
+        #extract the graph variables table so we can verify a variable has been added to the list
+        graph_vars_table = self.variable_manager.findChild(QtWidgets.QTableView, GRAPH_VARIABLES_QT)
+        expected_number_of_variables = graph_vars_table.model().rowCount(QtCore.QModelIndex()) + 1
+
         # Select variable type
         table_view = self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT)
         variable_entry = pyside_utils.find_child_by_pattern(table_view, new_variable_type)
 
-        # Click on it to create variable
+        # Click on it to create variable and wait for the UI to respond
         pyside_utils.item_view_index_mouse_click(table_view, variable_entry)
+        actual_number_of_variables = graph_vars_table.model().rowCount(QtCore.QModelIndex())
+        helper.wait_for_condition(lambda: (actual_number_of_variables == expected_number_of_variables),
+                                  WAIT_TIME_SEC_3)
```

Comment:
This function does 3 things when the name implies one thing. Let's extract those extra things.

Reply:
renamed the function and updated the comments to reflect what's going on. I think making a helper function for a helper function would be overdoing it in this case since some of the extra stuff being done is just gathering info for comparison (verification).